lib/carrierwave/base64/base64_string_io.rb

```diff
@@ -19,7 +20,7 @@ def initialize(encoded_file, file_name)
       end
 
       def original_filename
-        File.basename("#{@file_name}.#{@file_extension}")
+        @requested_filename || File.basename("#{@file_name}.#{@file_extension}")
```

Comment:
That won't work. The user might specify the filename, but without the extension.